misc/testerina/modules/testerina-runtime/src/main/java/org/ballerinalang/test/runtime/util/TesterinaUtils.java

```diff
@@ -123,92 +110,48 @@ private static void execute(TestSuite suite, ClassLoader classLoader, TestArgume
         }
-        Class<?> configClazz;
-        String configClassName = TesterinaUtils
-                .getQualifiedClassName(suite.getOrgName(), suite.getTestPackageID(), suite.getVersion(),
-                        CONFIGURATION_CLASS_NAME);
-        try {
-            configClazz = classLoader.loadClass(configClassName);
-        } catch (Throwable e) {
-            throw new BallerinaTestException("failed to load configuration class :" + configClassName);
-        }
-        String suiteExecuteFilePath = suite.getExecuteFilePath();
-        if (suite.getOrgName().equals(ANON_ORG) && suite.getTestPackageID().equals(DOT)) {
-            suiteExecuteFilePath = suiteExecuteFilePath.replace(DOT, FILE_NAME_PERIOD_SEPARATOR);
-        }
-        String testExecuteClassName = TesterinaUtils.getQualifiedClassName(suite.getOrgName(),
-                suite.getTestPackageID(),
-                suite.getVersion(),
-                suiteExecuteFilePath);
-        Class<?> testExecuteClazz;
-        try {
-            testExecuteClazz = classLoader.loadClass(testExecuteClassName);
-        } catch (Throwable e) {
-            throw new BallerinaTestException("failed to load test execution class :" + testExecuteClassName);
-        }
-        Scheduler scheduler = new Scheduler(4, false);
-        Scheduler initScheduler = new Scheduler(4, false);
-
-        // start TRAP signal handler which produces the strand dump
-        startTrapSignalHandler();
 
         // This will init and start the test module.
-        startSuite(suite, initScheduler, initClazz, configClazz, testExecuteClazz, args);
-        // Call module stop and test stop function
-        stopSuite(scheduler, initClazz);
+        startSuite(initClazz, args);
+        return getTestExecutionState(initClazz);
     }
 
-    private static void startSuite(TestSuite suite, Scheduler initScheduler, Class<?> initClazz,
-                            Class<?> configClazz, Class<?> testExecuteClazz, TestArguments args) {
-        TesterinaFunction init = new TesterinaFunction(initClazz, INIT_FUNCTION_NAME, initScheduler);
-        TesterinaFunction start = new TesterinaFunction(initClazz, START_FUNCTION_NAME, initScheduler);
-        TesterinaFunction configInit = new TesterinaFunction(configClazz, "$configureInit", initScheduler);
-        TesterinaFunction testExecute = new TesterinaFunction(testExecuteClazz, "__execute__", initScheduler);
-        // As the init function we need to use $moduleInit to initialize all the dependent modules
-        // properly.
-
-        Object response = configInit.directInvoke(new Class[]{String[].class, Path[].class, String.class},
-                new Object[]{new String[]{}, getConfigPaths(suite), null});
+    private static void startSuite(Class<?> initClazz, String[] args) {
+        // Call test module main
+        String funcName = cleanupFunctionName("main");
+        Object response = runTestModuleMain(initClazz, funcName, args, String[].class);
```

Comment:
Shall we pass a constant as the `funcName` whose value is "main" (Maybe already available), and we can remove the line `String funcName = cleanupFunctionName("main");`